shell-script/public-gateway-checker.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+# IPFS public gateway checker v1.0 (shell script version)
+# original repository: https://github.com/ipfs/public-gateway-checker
+# this script added by Joss Brown: https://github.com/JayBrown
+
+JSON_URL="https://raw.githubusercontent.com/ipfs/public-gateway-checker/master/gateways.json"
+TESTHASH="Qmaisz6NMhDB51cCvNWa1GMS7LU1pAxdF4Ld6Ft9kZEP2a"
+
+echo "Public IPFS Gateways"
+
+GATEWAYS=$(curl --silent "$JSON_URL" | sed -e '$ d' -e '1,1d' -e 's/\"//g')
+if [[ $GATEWAYS == "" ]] ; then
+	echo "Error: no gateways found."
+	exit
+fi
+
+GWNUM=$(echo "$GATEWAYS" | wc -l | xargs)
+
+while read -r GATEWAY
+do
+
+	GATEWAY=$(echo "$GATEWAY" | xargs)
+	TESTURL=$(echo "$GATEWAY" | sed -e "s-:hash,-$TESTHASH-" -e "s-:hash-$TESTHASH-")
+	DOMAIN=$(echo "$GATEWAY" | awk -F/ '{print $1"//"$3}')
+
+	if [[ $(curl --silent "$TESTURL") == "Hello from IPFS Gateway Checker" ]] ; then
```

Comment:
@VictorBjelkholm @koalalorenzo 

You can **dismiss this pull request**. I have a new cli here:

https://github.com/JayBrown/public-gateway-checker/blob/master/cli/ipfg

It should be tested first, before I create a new pull requests. I haven't yet had the time to test it thoroughly. But there's no rush anyway.

Changes:

```
renamed to ipfg
shebang with env bash
on 301-redirects etc. there's now a secondary curl with follow-redirect if the first curl returns an offline
invisible directory: $HOME/.ipfg
local backup of remote gateway list incl. update option -s
local gateway list with user-specified gateway urls
various options to add and remove gateway urls to/from the local list
options to check only remote gateways, only local gateways, or all gateways
option to manually check gateways with specified urls as args
options to only list (without checking) gateway urls (all, remote, local), either as domain or as raw url
miscellaneous options for: help page, print version, open GitHub repo
```